Interface.py

```diff
@@ -75,19 +79,33 @@ def shutdown(self, wait = True):
 #the future object is responsible for getting inquiries from the client about the server
 #and should keep all the info for the server and name, and facilitate communication between server and client
 class Future():
-    def __init__(self, function_name, server_info, func_args, func_kwargs):
+    def __init__(self, function_name, server_info, memory, priority, retries, *args, **kwargs):

```

Comment:
keep argument's name consistent with variable name.
instead of server_info, keep a reference to the Executor object.
I don't think you need any other parameters here.